File: app/src/main/java/com/mapmyindia/sdk/demo/java/model/GeoAnalyticsModel.java
package com.mapmyindia.sdk.demo.java.model;

import com.mapmyindia.sdk.geoanalytics.MapmyIndiaGeoAnalyticsRequest;
import com.mapmyindia.sdk.geoanalytics.MapmyIndiaGeoAnalyticsType;

/**
 * * Created by Saksham on 02-06-2021.
 **/
public class GeoAnalyticsModel {

    private MapmyIndiaGeoAnalyticsType type;
    private MapmyIndiaGeoAnalyticsRequest params;
    private String geoboundType;
    private String[] geoBound;

    public GeoAnalyticsModel(MapmyIndiaGeoAnalyticsType type, MapmyIndiaGeoAnalyticsRequest params, String geoboundType, String[] geoBound) {
        this.type = type;
        this.params = params;
        this.geoboundType = geoboundType;
        this.geoBound = geoBound;
    }

    public MapmyIndiaGeoAnalyticsType getType() {
        return type;
    }

    public MapmyIndiaGeoAnalyticsRequest getParams() {
        return params;
    }

    public String getGeoboundType() {
        return geoboundType;
    }

    public String[] getGeoBound() {
        return geoBound;
    }
}

File: app/src/main/java/com/mapmyindia/sdk/demo/kotlin/model/GeoAnalyticsModel.kt
package com.mapmyindia.sdk.demo.kotlin.model

import com.mapmyindia.sdk.geoanalytics.MapmyIndiaGeoAnalyticsRequest
import com.mapmyindia.sdk.geoanalytics.MapmyIndiaGeoAnalyticsType


/**
 * Created by Saksham on 27-09-2021
 */

data class GeoAnalyticsModel(
    val type: MapmyIndiaGeoAnalyticsType,
    val params: MapmyIndiaGeoAnalyticsRequest,
    val geoboundType: String,
    val geoBound: Array<String>
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as GeoAnalyticsModel

        if (type != other.type) return false
        if (params != other.params) return false
        if (geoboundType != other.geoboundType) return false
        if (!geoBound.contentEquals(other.geoBound)) return false

        return true
    }

    override fun hashCode(): Int {
        var result = type.hashCode()
        result = 31 * result + params.hashCode()
        result = 31 * result + geoboundType.hashCode()
        result = 31 * result + geoBound.contentHashCode()
        return result
    }
}